tldraw/tldraw
# (1/2) Timeout collaborator cursors

This PR adds a timeout to collaborator cursors.

It's part 1 of two PRs. The second one is smaller: https://github.com/tldraw/brivate/pull/2053

# What is this?

After three seconds of inactivity, collaborator cursors disappear.

![2023-06-08 at 10 42 43 - Moccasin Flamingo](https://github.com/tldraw/tldraw/assets/15892272/93e463aa-0329-4ecb-ada1-4c38b36a655b)

If you're following someone, you can always see their cursor.

![2023-06-08 at 10 45 42 - Olive Crayfish](https://github.com/tldraw/tldraw/assets/15892272/11e8d85a-18a8-4976-85c5-d14f3841c296)

# Is there anything else?
The PR also adds support for the brivate PR: https://github.com/tldraw/brivate/pull/2053

# Admin

### Change Type


### Test Plan

You probably need to test this locally, as we don't do multiplayer previews on this repo yet.
1. Open the same shared project in two browser sessions.
2. Move around the cursor in one session, while able to see it from the other.
3. Stop moving the cursor.
4. Make sure that the cursor disappears on the other session after 3 seconds.
5. Move the cursor again, and make sure it reappears it.
6. Make sure that viewport-following the user makes the cursor show permanently.

### Release Notes

- Brought back cursor timeouts. Collaborator cursors now disappear after 3 seconds of inactivity.
